packages/git/src/node/dugite-git.ts

```diff
@@ -287,6 +287,52 @@ export class DugiteGit implements Git {
         return this.nameStatusParser.parse(repository.localUri, result.stdout.trim().split('\0').filter(item => item && item.length > 0));
     }
 
+    async log(repository: Repository, options?: Git.Options.Log): Promise<CommitFragment[]> {
+        const commits: CommitFragment[] = [];
+        // If remaining commits should be calculated by the backend, then run `git rev-list --count ${fromRevision | HEAD~fromRevision}`.
+        // How to use `mailmap` to map authors: https://www.kernel.org/pub/software/scm/git/docs/git-shortlog.html.
+        // (Probably, this would the responsibility of the `GitHub` extension.)
+        const args = [
+            'log'
+        ];
+        if (options && options.branch) {
+            args.push(options.branch);
+        }
+        if (options) {
+            const range = this.mapRange(options.range);
+            args.push(...[range, '-C', '-M', '-m']);
+        }
+        const maxCount = options && options.maxCount ? options.maxCount : 0;
+        if (Number.isInteger(maxCount) && maxCount > 0) {
+            args.push(...['-n', `${maxCount}`]);
+        }
+        args.push(...['--name-status', '--date=unix', `--format=%n%n%H%n%aE%n%aN%n%ad%n%ar%n%s`, '-z', '--']);
+        if (options && options.uri) {
+            const file = Path.relative(repository.localUri, options.uri);
```

Comment:
I think this will not work on Windows. You cannot just use `node's` path to relativize the path, we need to convert to FS paths first.